x-pack/plugin/deprecation/src/test/java/org/elasticsearch/xpack/deprecation/ClusterDeprecationChecksTests.java

```diff
@@ -86,6 +86,82 @@ public void testUserAgentEcsCheck() {
         assertEquals(singletonList(expected), issues);
     }
 
+
+    public void testPipelinesWithJodaDeprecatedFormats() {
+        PutPipelineRequest pipelineRequest = new PutPipelineRequest("testingestpipeline",
+            new BytesArray("{\n" +
+                "  \"description\": \"Pipeline with date and date_index_name processor\",\n" +
+                "  \"processors\": [\n" +
```

Comment:
Maybe add a `date` processor to this that doesn't use deprecated formats to help make this more resistant to regressions.